### PR TITLE
feat(ingest/unity): Ingest ownership for containers; lookup service principal display names

### PR DESCRIPTION
**Pull request by asikowitz**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py [hunk 1]
```diff
@@ -268,10 +279,11 @@ def process_table(
 
         ownership = self._create_table_ownership_aspect(table)
 
+        lineage: Optional[UpstreamLineageClass] = None
         if self.config.include_column_lineage:
             self.unity_catalog_api_proxy.get_column_lineage(table)
             lineage = self._generate_column_lineage_aspect(dataset_urn, table)
-        else:
+        elif self.config.include_table_lineage:
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/unity/source.py:286]: This seemed like a bug -- we only want to ingest lineage if this is set

File: metadata-ingestion/src/datahub/ingestion/source/unity/config.py [hunk 2]
```diff
@@ -50,9 +50,9 @@ class UnityCatalogSourceConfig(StatefulIngestionConfigBase, DatasetSourceConfigM
         description="Option to enable/disable lineage generation.",
     )
 
-    include_table_ownership: bool = pydantic.Field(
+    include_ownership: bool = pydantic.Field(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/unity/config.py:53]: let's add a `pydantic_field_renamed` helper to preserve backwards compatibility

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py:291]: yup looks good

**Review by hsheth2 — APPROVED** [review]
LGTM

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py (hunk 1 above)
Comment (shirshanka) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py:291]: nice catch

File: metadata-ingestion/src/datahub/ingestion/source/unity/config.py (hunk 2 above)
Comment (shirshanka) [reply metadata-ingestion/src/datahub/ingestion/source/unity/config.py:53]: +1 on that. Will merge after that

**Pull request merged**